src/myrtlespeech/data/sampler.py

```diff
@@ -56,23 +56,12 @@ def _batch_indices(self, indices, batch_size, drop_last):
         return batches
 
     def __iter__(self):
-        if self._n_iterators in self._sequential:
-            iter_ = self._seq_iter()
-        else:
-            iter_ = self._rnd_iter()
-        self._n_iterators += 1
-        return iter_
-
-    def _seq_iter(self):
-        for b in self.batch_indices:
-            yield b
-
-    def _rnd_iter(self):
         indices = list(range(len(self.batch_indices)))
-        if self.shuffle:
+        if self.shuffle and self._n_iterators not in self._sequential:
             random.shuffle(indices)
         for index in indices:
             yield self.batch_indices[index]
+        self._n_iterators += 1
```

Comment:
This should be moved before the for loop to match the semantics described in the comment. For instance, consider the following:

```python
indices = list(range(64))

srs = SequentialRandomSampler(
    indices=indices,
    batch_size=8,
    shuffle=True,
    sequential={0}
)

iter_1 = iter(srs)
iter_2 = iter(srs)

print(next(iter_1))
print(next(iter_2))
```

This will output:

```
[0, 1, 2, 3, 4, 5, 6, 7]
[0, 1, 2, 3, 4, 5, 6, 7]
```

When it should be `[0, 1, 2, 3, 4, 5, 6, 7]` followed by a random batch of indices?

Reply:
Done it!